go-controller/pkg/clustermanager/routeadvertisements/controller.go

```
@@ -586,6 +541,198 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
+			continue
+		}
+
+		nads, err := c.getSelectedNADs(otherRA.Spec.NetworkSelectors)
+		if err != nil {
+			return err
+		}
+		if len(nads) == 0 {
+			return fmt.Errorf("%w: failed checking cross RA overlap with RouteAdvertisement %q: no networks selected", errPending, ra.Name)
+		}
+		otherRASelectedNetworks, err := c.getSelectedNetworkInfo(nads, sets.New(otherRA.Spec.Advertisements...))
+		if err != nil {
+			// If we can't get networks from another RA, log but don't fail
+			// This could happen if the other RA has configuration issues
+			klog.V(4).Infof("Failed to get selected networks from RouteAdvertisement %q for cross overlap check with RouteAdvertisement %q: %v", otherRA.Name, ra.Name, err)
+			continue
+		}
+
+		otherVRFSubnets := createNetworkVRFsSubnetsMap(otherRA.Spec.TargetVRF, otherRASelectedNetworks)
+		otherSubnets, hasMatchingVRF := otherVRFSubnets[targetVRF]
+		if !hasMatchingVRF || len(otherSubnets) == 0 {
+			continue
+		}
+
+		// Check for overlaps between current RA subnets and other RA subnets in the same VRF
+		for _, currentSubnet := range currentRASubnets {
+			for _, otherSubnet := range otherSubnets {
+				overlap, err := subnetsOverlap(currentSubnet, otherSubnet)
+				if err != nil {
+					return fmt.Errorf("failed to check subnet overlap between %s and %s: %w", currentSubnet, otherSubnet, err)
+				}
+				if overlap {
+					subnetsByRA := map[string]string{
+						ra.Name:      currentSubnet,
+						otherRA.Name: otherSubnet,
+					}
+					raNames := []string{ra.Name, otherRA.Name}
+					slices.Sort(raNames)
+					conflictErr := fmt.Errorf("%w: overlapping CIDR detected between RouteAdvertisements %q and %q in VRF %q: [%s %s]",
+						errConfig, raNames[0], raNames[1], targetVRF, subnetsByRA[raNames[0]], subnetsByRA[raNames[1]])
+
+					if err := c.updateRAStatus(otherRA, false, conflictErr); err != nil {
+						klog.Errorf("Failed to update status of conflicting RouteAdvertisements %q: %v", otherRA.Name, err)
+					}
```

Are we sure we don't want to return the error so we retry the ra status update ?